WSLBlobNFS/WSLBlobNFS.psm1

```diff
@@ -6,178 +11,539 @@ Set-PSDebug -Strict
+    }
+    else
+    {
+        $initialized = $true
+        if($Force)
+        {
+            Write-Debug "Force parameter is provided. Copying files again."
+        }
+        elseif ($LASTEXITCODE -ne 0)
+        {
+            Write-Debug "Module is not installed. Copying files again."
+        }
+        else
+        {
+            Write-Debug "Module version is changed. Copying files again."
+        }
+
+        Copy-WSLFiles
+        if ($LASTEXITCODE -ne 0)
+        {
+            Write-Error "Copying wsl files failed."
+            return
+        }
+
+        Execute-WSL "echo $moduleVersionOnWin > $linuxScriptsPath/$versionFileName"
     }
 
-    Write-Host "Setting up the WSL environment for your WSLBlobNFS usage."
+    # Install systemd, restart wsl and install NFS & Samba
+    Execute-WSL "systemctl list-unit-files --type=service | grep systemd > /dev/null 2>&1"
+    if($LASTEXITCODE -eq 0 -and !$Force)
+    {
+        Write-Debug "Systemd is already installed. Skipping systemd installation."
+    }
+    else
+    {
+        $initialized = $true
+        if($Force)
+        {
+            Write-Debug "Force parameter is provided. Installing systemd again."
+        }
+        else
+        {
+            Write-Debug "Systemd is not installed. Installing systemd."
+        }
+
+        Execute-WSL "$linuxScriptsPath/$wslScriptName installsystemd"
+        Write-Verbose "Installed systemd."
+
+        # Shutdown wsl and it will restart with systemd on next wsl command execution
+        wsl -d $distroName --shutdown
+    }
 
     # wsl shutsdown after 8 secs of inactivity. Hence, we need to run dbus-launch to keep it running.
     # Check the issue here:
     # https://github.com/microsoft/WSL/issues/10138
+    # To-do: Check if dbus is already running or not.
+    # dbus[488]: Unable to set up transient service directory: XDG_RUNTIME_DIR "/run/user/0/" is owned by uid 1000, not our uid 0
+    # Even if the systemd is installed, we need to run dbus-launch to keep it running.
+    wsl -d $distroName --exec dbus-launch true
 
-    # To-do: Copy only if the file is not present or if the file is modified.
-    # Files saved from windows will have \r\n line endings. Hence, we need to remove \r.
-    wsl -d Ubuntu-22.04 -u root -e bash -c "mkdir -p /root/scripts"
+    Execute-WSL "dpkg -s nfs-common samba > /dev/null 2>&1"
+    if($LASTEXITCODE -eq 0 -and !$Force)
+    {
+        Write-Debug "NFS & Samba are already installed. Skipping their installation."
+    }
+    else
+    {
+        $initialized = $true
+        if (!$Force)
+        {
+            Write-Debug "Force parameter is provided. Installing NFS & Samba again."
+        }
+        else
+        {
+            Write-Debug "NFS & Samba are not installed. Installing NFS & Samba."
+        }
+        Execute-WSL "'$linuxScriptsPath/$wslScriptName' installnfssmb $smbUserName"
+        Write-Verbose "Installed NFS & Samba."
```

Comment:
Write-Success "NFS and Samba installed successfully!"

Reply:
I've the following format everywhere "Installing <package>" and "Installed <package>", so will stick to "Installed <package> successfully"?